### PR TITLE
feat: full workout history data export (CSV + JSON)

### LINKED ISSUE #90: feat: full workout history data export (CSV)
## Context

Requested by u/Pretty-Dimension-213 on [r/tonalgym launch thread](https://www.reddit.com/r/tonalgym/):

> Do you have an easy way to help get someone's whole workout history? Like just into a csv

## Problem

Users have no way to export their Tonal workout data in a portable format. The app syncs and displays the data but doesn't offer a download/export option.

## Proposal

Add a data export feature (likely in Settings) that lets users download their full workout history as a CSV. Could include:

- Date, workout name, target area, duration, total volume
- Per-exercise breakdowns (exercise name, sets, reps, weight)
- Strength score history over time

## Notes

- The data is already available via the Tonal API proxy and partially persisted in the DB from history sync.
- CSV is the most universally useful format. JSON could be offered as a secondary option for technical users.
- Related to the existing `DataExport` settings component (`src/features/settings/DataExport.tsx`) — may already have scaffolding for this.

### PR DESCRIPTION
**Pull request by JeffOtano**
## Summary

- Add CSV export for workout history, exercise details, strength scores, and external activities alongside the existing JSON export
- Extend the data export to include all persisted Tonal data: external activities, current strength scores, and muscle readiness
- Fetch fresh workout history from the Tonal API (limit=500) during export to fill gaps beyond the 100-workout backfill window

Closes #90

## Changes

**Backend (`convex/dataExport.ts` — new file)**
- Extracted export logic from `account.ts` into its own module (one-concept-per-file)
- `collectUserData` query now fetches 10 tables: completedWorkouts, exercisePerformance, strengthScoreSnapshots, currentStrengthScores, muscleReadiness, externalActivities, workoutPlans, weekPlans, checkIns, and userProfiles
- Movement names resolved from the `movements` catalog so exercise CSVs show human-readable names
- `exportData` action fetches fresh Tonal workout history (limit=500, single API call) and merges with DB records, deduplicating by activityId
- `getKnownActivityIds` internal query supports the dedup logic

**Frontend (`src/features/settings/DataExport.tsx`)**
- Added format selector dropdown with 5 export options: All Data (JSON), Workout History (CSV), Exercise Details (CSV), Strength Scores (CSV), External Activities (CSV)
- Each format uses an appropriate MIME type and descriptive filename

**CSV utilities (`src/features/settings/csvExport.ts` — new file)**
- `workoutsToCsv` — date, workout name, target area, duration, volume, work, type
- `exercisesToCsv` — date, exercise name, movement ID, sets, reps, avg weight, volume
- `strengthScoresToCsv` — date, overall/upper/lower/core scores
- `externalActivitiesToCsv` — time, type, source, duration, calories, heart rate, distance
- Proper CSV escaping for commas, quotes, and newlines

**Cleanup (`convex/account.ts`)**
- Removed export-related code (interface, action, query) — now lives in `dataExport.ts`
- Removed unused `analytics` import

## Checklist

- [x] `npx tsc --noEmit` passes (pre-existing convex module errors only; no new errors in changed files)
- [x] `npm run lint` passes
- [x] `npm test` passes (existing + new tests) — 79 files passed, 838 tests passed; 3 pre-existing infra errors from missing `@edge-runtime/vm` unrelated to this PR
- [x] New logic has tests (happy path + at least one error/edge case)
- [x] No new or materially expanded file exceeds the 300-line soft cap — `dataExport.ts` is 301 lines (sequential coordination query); all other files well under
- [x] No file exceeds the 400-line hard cap (enforced by CI)
- [x] Functions stay near the 60-line convention — `collectUserData` is 162 lines but is a sequential coordination function (imperative shell) that queries 10 tables and maps results; per AGENTS.md guidance, kept together rather than artificially split
- [x] No new `any` types (enforced by ESLint); `as` casts only at deserialization boundaries
- [ ] Tested in browser (if UI changes) — cannot start dev server in this environment; format selector and CSV download logic tested via unit tests
- [x] Commits follow conventional format (`type: description`)
- [x] No unused exports or dead code introduced (`npx knip` clean)

## Test Plan

- **21 unit tests** covering all new code:
  - `csvExport.test.ts` (10 tests): empty input headers, row conversion, CSV escaping for commas/quotes, null value handling for each converter
  - `DataExport.test.tsx` (11 tests): format selector rendering, JSON download on default format, CSV download for each of the 4 CSV formats, loading/disabled state, error handling, button re-enable after export
- **Lint, format, knip, and file-size checks** all pass locally
- Tonal API enrichment uses a single API call (limit=500) wrapped in try/catch — gracefully falls back to DB-only data if the API is unavailable

https://claude.ai/code/session_019XikmcHgT71BmQjtDQUYxR

## Summary by CodeRabbit

* **New Features**
  * Export your data in multiple formats: JSON or CSV (workouts, exercises, strength scores, activities).
  * Exports now include fresh Tonal workout history when your account is connected.

* **Tests**
  * Added comprehensive test coverage for data export and CSV formatting.

### CODE REVIEW

File: convex/dataExport.ts [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+  return {
+    date: a.activityTime?.split("T")[0] ?? "",
+    title: p?.workoutTitle ?? "Unknown",
+    targetArea: p?.targetArea ?? "",
+    totalDuration: p?.totalDuration ?? 0,
+    totalVolume: p?.totalVolume ?? 0,
+    totalWork: p?.totalWork ?? 0,
+    workoutType: p?.workoutType ?? "",
+  };
+}
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply convex/dataExport.ts:85]: Fixed in 162c123 — removed the unnecessary `?.` since `Activity.activityTime` is `string` (required).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: convex/dataExport.ts [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+
+export const exportData = action({
+  args: {},
+  handler: async (ctx): Promise<ExportedData> => {
+    const userId = await ctx.runQuery(internal.lib.auth.resolveEffectiveUserId, {});
+    if (!userId) throw new Error("Not authenticated");
+
+    const data = (await ctx.runQuery(internal.dataExport.collectUserData, {
+      userId,
+    })) as ExportedData;
```
[the remark this answers is not in the document]
Comment (JeffOtano) [reply convex/dataExport.ts:95]: Fixed in 4525c65 — manually added `dataExport` to `convex/_generated/api.d.ts` (import + module map entry). `npx convex codegen` requires a network connection to the Convex deployment which wasn't available, so the generated file was updated by hand following the same alphabetical pattern.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by JeffOtano** [comment]
## Responses to nitpick review comments

### 1. Add auth comment to `collectUserData` (`dataExport.ts:140`)

Skipping — `collectUserData` is an `internalQuery`, which by definition is only callable by other Convex functions, never from the frontend. Every `internalQuery` in the codebase follows this same contract (e.g., `accountDeletion.deleteAllUserData`, `account.getUserEmail`). Per AGENTS.md: "Comments explain WHY, never WHAT" — a comment restating the standard `internalQuery` contract would be explaining what every maintainer already knows from the function registration type.

### 2. Remove `as Promise<...>` casts on `ctx.runQuery`/`ctx.runAction` (`dataExport.ts:102-108`)

Skipping — the suggestion to use generics like `ctx.runQuery<string[]>()` isn't possible. Convex's `ctx.runQuery` signature is:

```typescript
runQuery: (query: FunctionReference<"query", "public" | "internal">, args?: Record<string, Value>) => Promise<any>;
```

There are no generic type parameters available. The `as Promise<T>` cast is the only way to type the result, and it's the established pattern throughout the codebase — see `weekPlanEnriched.ts:176` and `coach/pushAndVerify.ts:84` for the same pattern.

### 3. Remove `as ExportedData` cast on `collectUserData` result (`dataExport.ts:93-95`)

Skipping — same root cause as #2. `ctx.runQuery` returns `Promise<any>` regardless of the handler's return type. Convex doesn't propagate handler return types through `internal.*` references. Adding Zod validation on output from our own internal function would contradict the project's convention: "Validate external input with Zod at the action boundary. Internal functions receive typed data" (AGENTS.md).

### 4. Extract typed handler for `<select>` onChange (`DataExport.tsx:116`)

Skipping — this is a one-line inline callback: `(e) => setFormat(e.target.value as ExportFormat)`. Per AGENTS.md: "Don't create helpers, utilities, or abstractions for one-time operations." The `as ExportFormat` cast is safe because the `<option>` values are generated from `Object.entries(FORMAT_LABELS)` which is `Record<ExportFormat, string>`, so every possible value is a valid `ExportFormat`.

---
_Generated by [Claude Code](https://claude.ai/code)_
